python/packages/ai/teams/ai/ai.py

```diff
@@ -336,3 +336,29 @@ async def _on_too_many_steps(
     ) -> str:
         self._logger.error("The run retrieval for the Assistants Planner has expired.")
         return ActionTypes.STOP
+
+    async def do_action(
+        self,
+        context: TurnContext,
+        state: StateT,
+        action: str,
+        parameters: Optional[Dict[str, Any]] = None
+    ) -> str:
+        """
+        Manually executes a named action.
+        
+        Args:
+            context (TurnContext): Current turn context.
+            state (StateT): Current turn state.
+            action (str): Name of the action to execute.
+            parameters (Optional[Dict[str, Any]]): Optional. Entities to pass to the action.
+        
+        Returns:
+            str: The result of the action.
+        """
+        if action not in self._actions:
+            raise ApplicationError(f"Can't find an action named '{action}'.")
+
+        action_entry = self._actions[action]
+        handler = action_entry.func
+        return await handler(ActionTurnContext(action, parameters, context), state)
```

Comment:
the `action_entry` class has a function called invoke that will call its respective handler

Reply:
can reference the `_on_do_command` to follow a similiar format